android/src/main/java/com/mattermost/networkclient/APIClientModule.kt

```diff
@@ -0,0 +1,102 @@
+package com.mattermost.networkclient
+
+import com.facebook.react.bridge.*
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import com.facebook.react.bridge.Arguments;
+
+class APIClientModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
+
+    var client = OkHttpClient();
+    var sessions = mutableMapOf<String, Request.Builder>()
+
+    override fun getName(): String {
+        return "APIClient"
+    }
+
+    @ReactMethod
+    fun createClientFor(baseUrl: String, config: ReadableMap, promise: Promise){
+        try {
+            sessions[baseUrl] = Request.Builder().url(baseUrl);
+            promise.resolve(sessions[baseUrl]);
+        } catch (err: Throwable){
+            promise.reject(err)
+        }
+    }
+
+    @ReactMethod
+    fun getClientHeadersFor(baseUrl: String, promise: Promise){
+        try {
+            val headersMap = Arguments.createMap()
+            sessions[baseUrl]!!.build().headers.forEach{(k,v) -> headersMap.putString(k, v)}
+            promise.resolve(headersMap)
+        } catch (error: Error) {
+            promise.reject(error)
+        }
+    }
+
+    @ReactMethod
+    fun addClientHeadersFor(baseUrl: String, headers: ReadableMap, promise: Promise){
+        for ((k,v) in headers.toHashMap()){
+            sessions[baseUrl]!!.addHeader(k,v as String);
+        }
+        val headersMap = Arguments.createMap()
```

Comment:
`promise.resolve()` does need a value, and while we are expecting `Promise<void>` on the App side - I had some feedback around returning the  empty `promise.resolve("")` so I'd rather return the headers for now; I think it may help on the App side as well to ensure the headers are set rather than setting and double checking.

Happy eitherway though :)